<?xml version="1.0"?>
<launch>

  <param name="use_sim_time" value="true"/>

  <node pkg="tf"  type="static_transform_publisher" name = "staticLaser" args = "-0.61 0.0 0.0 0.0 0.0 0.0  laser base_footprint 20"/>
  <node pkg="tf"  type="static_transform_publisher" name = "staticImu" args = "0.0 0.0 0.0 0.0 0.0 0.0  base_footprint_ekf odom_combined 20"/>
  <node pkg="tf"  type="static_transform_publisher" name = "staticMap" args = "0.0 0.0 0.0 0.0 0.0 0.0  map0 map 20"/>

    <include file="$(find ohm_pf)/launch/ohm_pf.launch">
      <arg name="output" value="screen" />
      <arg name="topOdometry" value="wheelodom" />
      <arg name="tfFixedFrame" value="map" />
      <arg name="tfBaseFootprintFrame" value="base_footprint" />
      <arg name="top2dPoseEst" value="initialpose" />
      <arg name="topCeilCam" value="pub_PoseTopic" />
      <arg name="topScan" default="filtered_scan" />
      <arg name="topMap" value="map" />
      <arg name="topMapSrv" value="static_map" />
      <arg name="samplesMax" value="10000" />
      <arg name="samplesMin" value="50" />
      <arg name="maxDistanceProbMap" value="3" />
      <arg name="subsamplingRateLaser" value="3" />
      <arg name="resamplingIntervallFilter" value="1" />
      <arg name="uncertaintyLaser" value="0.5" />
      <arg name="resamplingMethod" default="LV" />
    </include>

    <node name="mapServer" pkg="map_server" type="map_server" args="$(find ohm_pf)/bags/mapKupfer.yaml" />

  <!--EKF -->
  <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
    <param name="output_frame" value="map"/>
    <param name="base_footprint_frame" value="base_footprint_ekf"/>
    <param name="freq" value="30.0"/>
    <param name="sensor_timeout" value="1.0"/>
    <param name="odom_used" value="true"/>
    <param name="imu_used" value="true"/>
    <param name="vo_used" value="false"/>
    <param name="gps_used" value="false"/>

    <remap from="odom" to="wheelodom" />
    <remap from="imu_data" to="imu_fused" />
  </node>

  <node pkg="rosbag" type="play" name="bagPlayer" args="--clock -s 15 -r 1.0 $(find ohm_pf)/bags/boschKupferspuleCeilCam.bag"/>

  <node pkg="rviz" type="rviz" name="rviz"/>

</launch>
